crates/router/src/core/payments.rs

```diff
@@ -4579,40 +4579,35 @@ where
     .await
     .change_context(errors::ApiErrorResponse::InternalServerError)?;
 
+    let connectors = routing::perform_eligibility_analysis_with_fallback(
```

Comment:
I see this is adding the fallback as well to the list and this list is getting sent to perform SR. We should only perform eligibility analysis on the connectors and send only these filtered connectors to perform SR instead of adding the fallback too. Fallback connectors should be appended to the list obtained after performing SR

Reply:
This will be addressed in this [PR](https://github.com/juspay/hyperswitch/pull/6381).